octue/cloud/credentials.py

```diff
@@ -1,61 +1,106 @@
 import json
 import logging
 import os
-
+import warnings
+from json.decoder import JSONDecodeError
+from google.auth import compute_engine
 from google.oauth2 import service_account
-
+from octue.exceptions import InvalidInputException
 
 logger = logging.getLogger(__name__)
 
+DEFAULT_ENVIRONMENT_VARIABLE_NAME = "GOOGLE_APPLICATION_CREDENTIALS"
+
 
 class GCPCredentialsManager:
     """A credentials manager for Google Cloud Platform (GCP) that takes a path to a service account JSON file, or a
     JSON string of the contents of such a service account file, from the given environment variable and instantiates
     a Google Cloud credentials object.
 
-    :param str|None environment_variable_name:
+    :param str environment_variable_name:
     :return None:
     """
 
-    def __init__(self, environment_variable_name="GOOGLE_APPLICATION_CREDENTIALS"):
+    def __init__(self, environment_variable_name=DEFAULT_ENVIRONMENT_VARIABLE_NAME):
+
+        if environment_variable_name is None:
+            raise InvalidInputException(
+                "Cannot accept `None` as an environment variable name. Either specify the name as a string or use the default value"
+            )
```

Comment:
Why can't it be `None` here? I added the option for a specific nuanced use case but can't remember exactly what it was